aip/general/0193.md

```diff
@@ -44,15 +44,23 @@ strings **may** change over time; however, if an error message does not have a
 machine-readable identifier _in addition to_ the `code` field, changing the
 error message **must** be considered a backwards-incompatible change.
 
-**Note:** The [`ErrorInfo`][ErrorInfo] message is the recommended way to send a
-machine-readable identifier.
+#### ErrorInfo
+
+The [`ErrorInfo`][ErrorInfo] message is the required way to send a
+machine-readable identifier. All error responses **must** include an
+`ErrorInfo` payload in the `details` field. Variable information
+**should** be included in the `metadata` field on `ErrorInfo` and
```

Comment:
```suggestion
**must** be included in the `metadata` field on `ErrorInfo` and
```

should feels like "there's good reasons not to". So must?

Reply:
It says, **must**, if it is the error message, in the next phrase. Should I remove the next phrase or change it to, "...especially **must**..."?

Reply:
One reason not to change this to **must** is that removing keys from `metadata` represents a breaking change, which is explained the next PR. So, service authors may want to consider whether or not to add variable information to `metadata` not found in error messages to the map. They may only wish to do so if it is critical to the client.

Reply:
re: redundant must: just remove the second one? The first must's scope should reach across the whole sentence.

regarding the latter comment: then why *should*, and why not *may*? Overall the problem with anything outside of the *must* category is that it's a little ambiguous as to what the guidance means.

I think it was fine to merge it in, it's a higher level problem that we don't have good definitions of must/should/should not /must not anyway.